scripts/ci/images/fedora-sanitizers/Dockerfile

```diff
@@ -0,0 +1,51 @@
+FROM fedora:latest
+
+# Install core dev packages
+RUN dnf -y install gcc gcc-c++ gcc-gfortran git make curl file patch \
+        zlib-devel bzip2 bzip2-libs bzip2-devel blosc-devel vim valgrind \
+        zeromq-devel libpng-devel hdf5-devel openmpi-devel hdf5-openmpi-devel \
+        python3-devel python3-numpy python3-mpi4py-openmpi \
+        libasan libtsan libubsan clang llvm-devel libcxx-devel
+
+# Cleanup headers and packages
+RUN dnf clean all
+
+# Install CMake nightly
+WORKDIR /opt/cmake
+RUN curl -L https://cmake.org/files/dev/cmake-3.16.20200111-g8b29a83-Linux-x86_64.tar.gz |tar --strip-components=1 -xzv
+ENV PATH=/opt/cmake/bin:${PATH}
+
+# Installing Kitware Ninja
+WORKDIR /usr/local/bin
+RUN curl -L https://github.com/Kitware/ninja/releases/download/v1.9.0.g99df1.kitware.dyndep-1.jobserver-1/ninja-1.9.0.g99df1.kitware.dyndep-1.jobserver-1_x86_64-linux-gnu.tar.gz | tar --strip-components=1 -xzv
+
+# Building an msan-instrumented libc++
+WORKDIR /opt/llvm
+RUN git clone --branch llvmorg-9.0.0 --depth 1 \
+        https://github.com/llvm/llvm-project.git source && \
+    mkdir -p build/libcxx-msan && \
+    cd build/libcxx-msan && \
+    export \
+        CC=/usr/bin/clang \
+        CXX=/usr/bin/clang++ && \
+    /opt/cmake/bin/cmake \
+        -GNinja \
+        -DCMAKE_BUILD_TYPE=Release \
+        -DCMAKE_INSTALL_PREFIX=/opt/llvm/9.0.0/libcxx-msan \
+        -DCMAKE_INSTALL_RPATH='$ORIGIN' \
+        -DLLVM_ENABLE_PROJECTS="libcxx;libcxxabi" \
+        -DLLVM_USE_SANITIZER=MemoryWithOrigins \
+        -DLIBCXX_ENABLE_ABI_LINKER_SCRIPT=OFF \
```

Comment:
What's the purpose of this and the rpath adjustment? It gives me the following error when I try to compile a minimal program:

```console
CMake Error at /opt/cmake/share/cmake-3.16/Modules/CMakeTestCXXCompiler.cmake:53 (message):
  The C++ compiler

    "/usr/bin/clang++"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: /root/build/CMakeFiles/CMakeTmp
    
    Run Build Command(s):/usr/local/bin/ninja cmTC_aa33a && [1/2] Building CXX object CMakeFiles/cmTC_aa33a.dir/testCXXCompiler.cxx.o
    clang-8: warning: argument unused during compilation: '-stdlib=libc++' [-Wunused-command-line-argument]
    [2/2] Linking CXX executable cmTC_aa33a
    FAILED: cmTC_aa33a 
    : && /usr/bin/clang++  -nostdinc++ -stdlib=libc++ -isystem /opt/llvm/9.0.0/libcxx-msan/include/c++/v1 -fsanitize=memory  -L /opt/llvm/9.0.0/libcxx-msan/lib -Wl,-rpath,/opt/llvm/9.0.0/libcxx-msan/lib CMakeFiles/cmTC_aa33a.dir/testCXXCompiler.cxx.o  -o cmTC_aa33a   && :
    clang-8: warning: argument unused during compilation: '-nostdinc++' [-Wunused-command-line-argument]
    /usr/bin/ld: /usr/lib64/clang/8.0.0/lib/linux/libclang_rt.msan_cxx-x86_64.a(ubsan_type_hash_itanium.cc.o): undefined reference to symbol '_ZTIN10__cxxabiv121__vmi_class_type_infoE'
    /usr/bin/ld: //opt/llvm/9.0.0/libcxx-msan/lib/libc++abi.so.1: error adding symbols: DSO missing from command line
    clang-8: error: linker command failed with exit code 1 (use -v to see invocation)
```

Reply:
msan stuff removed